### PR TITLE
sql: correct schema name for crdb_internal.tables

### LINKED ISSUE #54976: sql: crdb_internal.tables always shows schema_name of `public`
```
root@:26257/defaultdb> create schema my_schema;
CREATE SCHEMA

Time: 150ms total (execution 49ms / network 101ms)

root@:26257/defaultdb> create table my_schema.tab(a int primary key);
CREATE TABLE

Time: 120ms total (execution 55ms / network 65ms)

root@:26257/defaultdb> select * from [show tables] where table_name = 'tab';
  schema_name | table_name | type  | estimated_row_count
--------------+------------+-------+----------------------
  my_schema   | tab        | table |                   0
(1 row)

Time: 24ms total (execution 24ms / network 0ms)

root@:26257/defaultdb> select database_name, schema_name, name from crdb_internal.tables where name = 'tab';
  database_name | schema_name | name
----------------+-------------+-------
  defaultdb     | public      | tab
(1 row)

Time: 8ms total (execution 8ms / network 0ms)
```

The `schema_name` in this case should be `my_schema`.

When fixing this, watch out for #54966 and make sure auto stats collection is working correctly.

### PR DESCRIPTION
**Pull request by otan**
Resolves #54976 
Based on top of #54981 

Release note (bug fix): Previously, all tables in any schema showed up
as "public" in the schema_name crdb_internal column. They now display
the correct schema.

Release note (sql change): Add the parent_schema_id field to
crdb_internal.tables.

### CODE REVIEW

**Review by RaduBerinde — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55085#-:-MIW4O617mc7jhCU44iu:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @rytaft)

File: pkg/sql/stats/automatic_stats.go [hunk 1]
```diff
@@ -326,7 +326,7 @@ func (r *Refresher) ensureAllTables(
 	getAllTablesQuery := fmt.Sprintf(
 		`
 SELECT table_id FROM crdb_internal.tables AS OF SYSTEM TIME '-%s'
-WHERE schema_name = 'public'
+WHERE database_name IS NOT NULL
```
Comment (solongordon) [inline pkg/sql/stats/automatic_stats.go:329]: Is it a problem for this to return tables in virtual schemas?

Comment (otan) [reply pkg/sql/stats/automatic_stats.go:329]: virtual schemas have a null database in this table.

**Review by solongordon — APPROVED** [review]

**Comment by otan** [comment]
bors r=solongordon

**Pull request merged**